moped-etl/prefect/moped_test_create_flow.py

```diff
@@ -244,71 +268,55 @@ def drain_service(slug):
     )
     deploy_activity_log = activity_log.create_activity_log_task(
         command=commission_activity_log_command
     )
 
-    ## Apply migrations
-
+    ## Apply Migrations, metadata and optional seed data
     graphql_endpoint = "https://" + migrations.get_graphql_engine_hostname(slug=slug)
 
     access_key = migrations.get_graphql_engine_access_key(slug=slug)
 
-    rm_clone = "rm -fr /tmp/atd-moped"
-    cleaned = migrations.remove_moped_checkout(command=rm_clone)
-
-    git_clone = f"git clone {GIT_REPOSITORY} /tmp/atd-moped"
-    cloned = migrations.clone_moped_repo(command=git_clone, upstream_tasks=[cleaned])
-
-    checkout_branch = migrations.get_git_checkout_command(slug=slug)
-    checked_out = migrations.checkout_target_branch(
-        command=checkout_branch, upstream_tasks=[cloned]
-    )
-
     metadata = "metadata"
 
     config = migrations.create_graphql_engine_config_contents(
         graphql_endpoint=graphql_endpoint,
         access_key=access_key,
         metadata=metadata,
-        checked_out_token=checked_out,
-    )
-
-    # what are these parentheses doing?
-    migrate_cmd = (
-        "(cd /tmp/atd-moped/moped-database; hasura --skip-update-check migrate apply;)"
-    )
-    migrate = migrations.migrate_db(
-        command=migrate_cmd, upstream_tasks=[config, graphql_endpoint_ready]
+        checked_out_token=git_repo_checked_out,
     )
 
-    consistency_cmd = "(cd /tmp/atd-moped/moped-database; hasura --skip-update-check metadata inconsistency status;)"
-    consistent_metadata = migrations.check_for_consistent_metadata(
-        command=consistency_cmd, upstream_tasks=[migrate]
-    )
-
-    settled_metadata = migrations.sleep_fifteen_seconds(
-        consistent_metadata=consistent_metadata
-    )
+    with case(do_migrations, True):
+        # what are these parentheses doing?
+        migrate_cmd = "(cd /tmp/atd-moped/moped-database; hasura --skip-update-check migrate apply; sleep 15;)"
+        migrate = migrations.migrate_db(
+            command=migrate_cmd, upstream_tasks=[config, graphql_endpoint_ready]
+        )
 
-    metadata_cmd = (
-        "(cd /tmp/atd-moped/moped-database; hasura --skip-update-check metadata apply;)"
-    )
-    metadata = migrations.apply_metadata(
-        command=metadata_cmd, upstream_tasks=[migrate, settled_metadata]
-    )
+        consistency_cmd = "(cd /tmp/atd-moped/moped-database; hasura --skip-update-check metadata inconsistency status;)"
+        consistent_metadata = migrations.check_for_consistent_metadata(
+            command=consistency_cmd, upstream_tasks=[migrate]
+        )
 
-    with case(use_seed_data, True):
-        apply_seed_cmd = (
-            "(cd /tmp/atd-moped/moped-database; hasura --skip-update-check seed apply;)"
+        # Should this sleep come out or should the bash sleeps be done like this?
+        settled_metadata = migrations.sleep_fifteen_seconds(
+            consistent_metadata=consistent_metadata
```

Comment:
This is another way to do the sleeps, and this "looks" more prefecty, but it's a lot of hoop to jump through for a sleep.

Reply:
I can come up with good arguments either way for how to handle the sleeps. Part of me thinks it is fine in bash because a task finishes when it finishes it. Part of me thinks that the delay is more hidden away that way 🤷 I know this isn't super helpful 😆